### PR TITLE
feat: add toggle workflow description

### PR DESCRIPTION
**Pull request by rahalan**
Add documentation to AVM how to use the toggle worklow

### CODE REVIEW

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:1]: Can we have a couple of screenshots to show the usage? E.g., add examples to disable all avm workflows except one, which will be the most common use case for module owners. 

I'd also suggest to leverage hugo language, adding notes and important notes to highlight, e.g., that the workflow only works in forks

Comment (rahalan) [reply docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:1]: Done. Please review again

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+You can find the workflow under [`.github/workflows/avm.platform.toggle-avm-workflows.yml`](https://github.com/Azure/bicep-registry-modules/blob/main/.github/workflows/avm.platform.toggle-avm-workflows.yml)
+
+# How it works
+
+Select the workflow from the list
+
+<img src="../../img/contribution/selectToggleWorkflows.png" width=100%>
+
+Run the workflow `avm.platform.toggle-avm-workflows` and set the following settings:
+- `targetState` to enable or disable workflows
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:25]:
```suggestion
- `Enable or disable workflows` to enable or disable workflows
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- `includePattern` set the RegEx which workflows are included
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:26]:
```suggestion
- `RegEx which workflows are included` include a specific set of workflows, using a RegEx.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- `excludePattern` set the RegEx which workflows are excluded
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:27]:
```suggestion
-  `RegEx which workflows are excluded` exclude a specific set of workflows, using a RegEx.
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+<img src="../../img/contribution/runToggleWorkflows.png" width=100%>
+
+# Typical use cases
+
+If all but one workflow should be disabled, set the following settings:
+- `targetState` to `disable`
+- `includePattern` to `avm\.(?:res|ptn)` (that is the default setting)
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:35]:
```suggestion
-  `RegEx which workflows are included` to `avm\.(?:res|ptn)` (that is the default setting)
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+- `includePattern` to `avm.res.compute.virtual-machine` (the name of your workflow, here as example the one for the virtual machine)
+
+If all workflows should be enabled, set the following settings:
+- `targetState` to `enable`
+- `includePattern` to `avm\.(?:res|ptn)` (that is the default setting)
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:40]:
```suggestion
- `RegEx which workflows are included` to `avm\.(?:res|ptn)` (that is the default setting)
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+- `includePattern` to `^$` (that is the default setting)
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:41]:
```suggestion
- `RegEx which workflows are excluded` to `^$` (that is the default setting)
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+- `excludePattern` set the RegEx which workflows are excluded
+
+<img src="../../img/contribution/runToggleWorkflows.png" width=100%>
+
+# Typical use cases
+
+If all but one workflow should be disabled, set the following settings:
+- `targetState` to `disable`
+- `includePattern` to `avm\.(?:res|ptn)` (that is the default setting)
+- `includePattern` to `avm.res.compute.virtual-machine` (the name of your workflow, here as example the one for the virtual machine)
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:36]:
```suggestion
- `RegEx which workflows are excluded` to `avm.res.compute.virtual-machine` (the name of your workflow, here as example the one for the virtual machine)
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+- `targetState` to enable or disable workflows
+- `includePattern` set the RegEx which workflows are included
+- `excludePattern` set the RegEx which workflows are excluded
+
+<img src="../../img/contribution/runToggleWorkflows.png" width=100%>
+
+# Typical use cases
+
+If all but one workflow should be disabled, set the following settings:
+- `targetState` to `disable`
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:34]:
```suggestion
- `Enable or disable workflows` to `Disable`
```

File: docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+- `includePattern` to `avm\.(?:res|ptn)` (that is the default setting)
+- `includePattern` to `avm.res.compute.virtual-machine` (the name of your workflow, here as example the one for the virtual machine)
+
+If all workflows should be enabled, set the following settings:
+- `targetState` to `enable`
```
Comment (eriqua) [inline docs/content/contributing/bicep/bicep-contribution-flow/enable-or-disable-workflows.md:39]:
```suggestion
- `Enable or disable workflows` to `Enable`
```

**Review by eriqua — APPROVED** [review]
:ok:

**Review by eriqua — CHANGES_REQUESTED** [review]
Looks great in content @rahalan. Could you please have a look at linter issues raised by the PR checks?

**Comment by rahalan** [comment]
> Looks great in content @rahalan. Could you please have a look at linter issues raised by the PR checks?

@eriqua fixed, except one: having a heading at the beginning. The other files in the same hierarchy level also don't have one

**Review by eriqua — APPROVED** [review]

**Comment by eriqua** [comment]
> > Looks great in content @rahalan. Could you please have a look at linter issues raised by the PR checks?
> 
> @eriqua fixed, except one: having a heading at the beginning. The other files in the same hierarchy level also don't have one

@rahalan @AlexanderSehr this looks good to me, but I don't remember what was decided around this linter check. Are we waiting for a linter fix that would skip this control? Should we merge nevertheless?

**Comment by AlexanderSehr** [comment]
> > > Looks great in content @rahalan. Could you please have a look at linter issues raised by the PR checks?
> > 
> > 
> > @eriqua fixed, except one: having a heading at the beginning. The other files in the same hierarchy level also don't have one
> 
> @rahalan @AlexanderSehr this looks good to me, but I don't remember what was decided around this linter check. Are we waiting for a linter fix that would skip this control? Should we merge nevertheless?

If I'm not mistaken, @rahalan was supposed to add an exception for this rule to the linter config file

File: docs/static/img/contribution/runToggleWorkflows.png
Comment (AlexanderSehr) [inline docs/static/img/contribution/runToggleWorkflows.png:1]: To resolve by @rahalan:

Just a comment: In general I'd recommend to rather draw boxes in e.g. paint than trying to draw arrows - at least in public-facing material. This is however just a recommendation so feel free to resolve.

**Comment by rahalan** [comment]
> > > > Looks great in content @rahalan. Could you please have a look at linter issues raised by the PR checks?
> > > 
> > > 
> > > @eriqua fixed, except one: having a heading at the beginning. The other files in the same hierarchy level also don't have one
> > 
> > 
> > @rahalan @AlexanderSehr this looks good to me, but I don't remember what was decided around this linter check. Are we waiting for a linter fix that would skip this control? Should we merge nevertheless?
> 
> If I'm not mistaken, @rahalan was supposed to add an exception for this rule to the linter config file

Done

**Comment by AlexanderSehr** [comment]
There seems to be one issue left with a missing empty line at the end of the markdown

File: docs/static/img/contribution/runToggleWorkflows.png
Comment (rahalan) [reply docs/static/img/contribution/runToggleWorkflows.png:1]: @AlexanderSehr done

**Comment by rahalan** [comment]
> There seems to be one issue left with a missing empty line at the end of a markdown (see logs)

@AlexanderSehr fixed

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**